### PR TITLE
chore: remove all remaining references to App Engine

### LINKED ISSUE #973: Migrate App Engine to Cloud Run
Per https://cloud.google.com/appengine/docs/flexible/cloud-run-for-gae-customers :

> Cloud Run is the latest evolution of Google Cloud Serverless, building on the experience of running App Engine for more than a decade. Cloud Run runs on much of the same infrastructure as App Engine standard environment, so there are many similarities between these two platforms.
> 
> Cloud Run is designed to improve upon the App Engine experience, incorporating many of the best features of both App Engine standard environment and App Engine flexible environment. Cloud Run services can handle the same workloads as App Engine services, but Cloud Run offers customers much more flexibility in implementing these services. This flexibility, along with improved integrations with both Google Cloud and third-party services, also enables Cloud Run to handle workloads that cannot run on App Engine.

We should look at migrating the osv.dev frontend off App Engine, which will also give us the benefit of better homogenising our infra. This has benefits for our deployment and monitoring, as the relevant App Engine bits (versions, cron jobs etc) all live in a separate place from Cloud Run and Kubernetes. 

One consideration is Datastore latency -- we should ensure that there's no negative impact there.

### PR DESCRIPTION
**Pull request by michaelkedar**
Fully closes #973
We've been off App Engine for a while now. Cleaned up all the stray references to App Engine:
- renamed `gcp/appengine` to `gcp/website`
- renamed the poetry environment `appengine-backend` to `website-backend` (you'll have to recreate the environment)
- renamed targets in Makefile `appengine` -> `website`
- moved datastore `index.yaml` to new directory `gcp/datastore`
- removed the `deployment/gae` directory
- removed App Engine-specific checks from website handlers.
  - Moved website health check endpoint from `/_ah/warmup` to `/healthz`
- Updated various docs to no longer refer to App Engine

### CODE REVIEW

**Comment by another-rex** [comment]
Nice, the only slightly odd thing is the datastore index is now under the website folder. Can that be moved into its own folder?

**Comment by michaelkedar** [comment]
> Nice, the only slightly odd thing is the datastore index is now under the website folder. Can that be moved into its own folder?

I've given it its own folder for now in `gcp/datastore`, though this might end up in terraform somewhere.

**Review by another-rex — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]
nice!!

**Review by hogo6002 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -24,12 +24,13 @@ consists of:
...
 | `docker/`       | CI docker files (`ci`, `deployment`, `terraform`) <br /> Workers for bisection and impact analysis (`worker`, `importer`, `exporter`, `alias`, `worker-base`) <br /> The determine version `indexer`<br /> `cron/` jobs for database backups and processing oss-fuzz records |
 | `docs/`         | Jekyll files for https://google.github.io/osv.dev/ <br /> `build_swagger.py` and `tools.go` |
 | `gcp/api`       | OSV API server files (including files for the local ESP server) <br /> protobuf files in `/v1`|
-| `gcp/appengine` | The backend of the osv.dev web interface, with the frontend in `frontend3` <br /> Blog posts (in `blog`) <br /> App Engine Cron Handlers (to be removed) <br /> The  datastore indexes file (`index.yaml`) |
+| `gcp/website  ` | The backend of the osv.dev web interface, with the frontend in `frontend3` <br /> Blog posts (in `blog`) |
 | `gcp/functions` | The Cloud Function for publishing PyPI vulnerabilities (maintained, but not developed) |
+| `gcp/datastore` | The datastore index file (`index.yaml`) |
```
Comment (cuixq) [inline README.md:33]: sort `gcp` folders alphabetically?

**Review by cuixq — APPROVED** [review]

**Review by andrewpollock — APPROVED** [review]
I'd love to see this dependency on https://osv.dev/public_keys/pypa better documented somewhere... I just followed the revision history and some GitHub issues around previous unintended breakage to the consuming source and I wasn't able to quickly determine what is being signed by this key... @oliverchang ?

**Pull request merged**